applications/nrf5340_audio/src/modules/sd_card_playback.c

```diff
@@ -0,0 +1,362 @@
+	k_sem_give(&m_sem_playback);
+}
+
+void sd_card_playback_lc3(char *filename)
+{
+	playback_file_format = LC3;
+	playback_file_name = filename;
+	k_sem_give(&m_sem_playback);
+}
+
+int sd_card_playback_mix_with_stream(void *const pcm_a, size_t pcm_a_size)
+{
+	int ret;
+	uint8_t pcm_b[pcm_frame_size];
+	size_t write_size = pcm_frame_size;
+
+	ret = sd_card_playback_buffer_set(pcm_b, &write_size);
+	if (ret < 0) {
+		LOG_ERR("Error when loading pcm data into buffer. Ret: %d", ret);
+		return ret;
+	}
+	if (write_size == pcm_frame_size) {
+		pcm_mix(pcm_a, pcm_a_size, pcm_b, pcm_frame_size, B_MONO_INTO_A_STEREO_L);
+	} else {
+		LOG_WRN("Size written to buffer does not equal the requested size");
+	}
+	return 0;
+}
+
+int sd_card_playback_init(void)
+{
+	int ret;
+
+	sd_card_playback_thread_id = k_thread_create(
+		&sd_card_playback_thread_data, sd_card_playback_thread_stack,
+		SD_CARD_PLAYBACK_STACK_SIZE, (k_thread_entry_t)sd_card_playback_thread, NULL, NULL,
+		NULL, K_PRIO_PREEMPT(4), 0, K_NO_WAIT);
+	ret = k_thread_name_set(sd_card_playback_thread_id, "sd_card_playback");
+	if (ret < 0) {
+		LOG_ERR("Failed");
+		return ret;
+	}
+	return 0;
+}
+
+/* Shell functions */
+static int cmd_play_wav_file(const struct shell *shell, size_t argc, char **argv)
+{
+	char file_loc[MAX_PATH_LEN + MAX_FILENAME_LEN] = "";
+
+	strcat(file_loc, playback_file_path);
+	strcat(file_loc, argv[1]);
+	sd_card_playback_wav(file_loc, 10, 16, SAMPLE_RATE_48K);
+	return 0;
+}
+
+static int cmd_play_lc3_file(const struct shell *shell, size_t argc, char **argv)
+{
+	char file_loc[MAX_PATH_LEN + MAX_FILENAME_LEN] = "";
+
+	strcat(file_loc, playback_file_path);
+	strcat(file_loc, argv[1]);
+	sd_card_playback_lc3(file_loc);
+	return 0;
+}
+
+static int cmd_change_dir(const struct shell *shell, size_t argc, char **argv)
+{
+	/* Remember to change this to make sure you dont navigate to a nonexistent dir */
+	if (argv[1][0] == '/') {
+		playback_file_path[0] = '\0';
+		shell_print(shell, "Current directory: root");
+	} else {
+		strcat(playback_file_path, argv[1]);
+		strcat(playback_file_path, "/");
+		shell_print(shell, "Current directory: %s", playback_file_path);
+	}
+	return 0;
+}
+
+/* This function needs a lot of stack size. Shell stack size has to be 4096 instead of 1024 bc of
+ * this function. The reason for this is probably the sd_card_list_files function
+ */
+static int cmd_list_files(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	char buf[LIST_FILES_BUF_SIZE];
+	size_t buf_size = LIST_FILES_BUF_SIZE;
+
+	ret = sd_card_list_files(playback_file_path, buf, &buf_size);
+	if (ret < 0) {
+		shell_print(shell, "Something went wrong!\n");
+		return ret;
+	}
+	shell_print(shell, "%s", buf);
+	return 0;
+}
+
+/* Creating subcommands (level 1 command) array for command "demo". */
+SHELL_STATIC_SUBCMD_SET_CREATE(
+	sd_card_playback_cmd,
+	SHELL_COND_CMD(CONFIG_SHELL, play_lc3, NULL, "Play lc3 file.", cmd_play_lc3_file),
```

Comment:
This is not strictly needed in this PR, but how much work would it be to have just a `play`, and based on the file name ending the system itself detects if it is a wav or LC3 file?

Reply:
Good question. I will take a look at it when I am done with the other stuff:)

Reply:
Any update on this? Though, I do not want to block the PR with this featurecreep.

Reply:
I have not done any work on this feature yet